app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import androidx.test.espresso.Espresso.pressBack
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import kotlinx.android.synthetic.main.mozac_ui_tabcounter_layout.view.*
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.helpers.AndroidAssetDispatcher
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestAssetHelper
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+class SmokeTest {
+    private val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    private lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val activityTestRule = HomeActivityTestRule()
+
+    @Before
+    fun setUp() {
+        mockWebServer = MockWebServer().apply {
+            setDispatcher(AndroidAssetDispatcher())
+            start()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun verifyTheNavToolBarButtonsTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        //  - Visit a URL, wait until it's loaded
+        homeScreen {
+            navigationToolbar {
+            }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+                verifyPageContent(defaultWebPage.content)
+
+                // Verify the items of Nav URL Bar
+                verifyNavURLBarItems()
+
+                // Verify 'x' button actions
+                verifyXButtonActions()
+                pressBack()
+
+                // Verify various items after returning back to the initial WebPage
+                verifyPageContent(defaultWebPage.content)
+                verifyNavURLBar()
+                clickTabCounter()
+
+                // Verify items on HomeScreen

```

Comment:
From Browser robot you should use the [openHomescreen](https://github.com/mozilla-mobile/fenix/blob/master/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt#L338) transition go to to Homescreen and then once there use its robots methods for checking that you are  there, going to other menus from there...otherwise you  are adding code from that robot to the BrowserRobot

Reply:
yeah got that. So, that might be the test failure in emulator may be?

Reply:
![image](https://user-images.githubusercontent.com/46127128/82670568-08187900-9c5b-11ea-9406-ca0d26e1c9ab.png)
Do you mean in this way?

Reply:
yes, something like that 👍

Reply:
But when function `clickTabCounter()` is used it will take us to homescreen automatically. Hence, again using transition` openHomeScreen` causing error as it is already taken to the homescreen on clicking the number of tabs button.

Reply:
:eyes: There has  to be a  way, I would  need to check, but can't you go using a transition directly?

Reply:
Yeah I couldn't find any :/

Reply:
So,  I tried that, and locally is working:
 ```
               verifyNavURLBar()
            }.openHomeScreen {
                verifyHomeScreen()
                verifyExistingTabList()
            }
```

I think the issue is in after the `pressBack()` that the test does not wait  enough until the website is loaded to continue.
To be sure, try adding `TestAssetHelper.waitingTime` after that pressBack() if test passes, we will try to  find a way to be sure the website is loaded.

Reply:
It's seems you missed out adding 'clickTabCounter' function🤔

Reply:
No, I didn't, it can go from navigation to homescreen if there is a transition defined...

Reply:
Oops,I should have gone through it carefully. Thanks for pointing out :)

Reply:
After pushing the code, test is failing again!